lib/stellar-core-backup/job.rb

```diff
@@ -81,11 +88,18 @@ def run()
             # using sudo, if running as non root uid then you will need to configure sudoers
             stop_core = @cmd.run_and_capture('sudo', ['/bin/systemctl', 'stop', 'stellar-core'])
             # only proceed if core is stopped and FS is clean
-            if stop_core.success then
-              @backup_archive = @s3.get(@s3.latest)
+            if stop_core.success
+              if @clean
+                `rm -fr /var/lib/stellar/buckets/*`
+              end
               if @fs_restore.core_data_dir_empty?() then
-                # proceed
-                @utils.restore(@backup_archive)
+                @backup_archive = @s3.get(@s3.latest)
+                @fs_restore.restore(@backup_archive)
```

Comment:
Should we have an `StellarCoreBackup::utils.extract_backup(@backup_archive)` method which calls the `StellarCoreBackup::Tar.unpack(backup_archive, @working_dir)` method ?

This way we can verify the backup archive before restoring the buckets directory and then we would also be able to use `StellarCoreBackup::utils.restore` wrapper for the actual restore.

Reply:
Will you be moving initial backup extraction to it's own method and verifying the checksums prior to restoring ?

Reply:
Addressed as enhancement in https://github.com/stellar/puppet-v4/issues/178

Reply:
The requested functionality is in place. Backup archives are extracted and then checked. The restore process stops on gpp or shasum failure.